src/java/org/apache/cassandra/service/accord/async/AsyncOperation.java

```diff
@@ -301,6 +300,13 @@ public void start(BiConsumer<? super R, Throwable> callback)
     {
         Invariants.checkState(this.callback == null);
         this.callback = callback;
+        if (commandStore.inStore())
```

Comment:
why this change?  personally I would just call `this.run()` for better maintain... for example what happens if `loader.load` fails?

Reply:
This was a necessary change to ensure the `CommandsForKey` pruned key loading works as intended - we don't intend to ever write to disk if we are fetching data from disk to update the `CommandsForKey`, so this change permits us to immediately take a reference to the data we will use, preventing it from being evicted before we first run and despatch any reads. It has the advantage of also being much more efficient, avoiding a whole additional queue delay and additional work saving and loading.

The logic is different to running however - we do not want to `run` because we haven't finished running the `AsyncOperation` we're already executing, and stopping part way to run another `AsyncOperation` (if everything is already loaded) is not only tricky book-keeping wise, but also logically confusing for the caller that expects this to run after the current operation completes.

Reply:
Fair point regarding execution failure though - I can refactor either to share some execution logic (e.g. with a run method that can accept a highest state to execute to), or else special-case this to replicate exit logic. Whichever you have a preference for.

Reply:
> That said, the load logic is mostly asynchronous, so we'd be talking about serious logic bugs I think.


this code path has broken in nightly benchmark clusters often... so making sure we properly handle errors is best...